### PR TITLE
✨ Update sidebar User Interface

### PR DESCRIPTION
**Pull request by Pinimes**
## Status
Ready

## Description
- Revise the function-oriented sidebar by incorporating thumbnails and enhancing its visual appeal.

## Screenshot

sidebar

### CODE REVIEW

File: components/tags.js [hunk 1]
```diff
@@ -1,33 +1,88 @@
...
+          const videoId = this.getFirstVideoIdForTag(tag);
+          console.log("Video Tag ID"+ videoId);
+          if (videoId) {
+            this.getYouTubeThumbnail(videoId, tag);
+          }
+          console.log("Video tag ID after "+tag);
+        });
+    }
+
+    getYouTubeThumbnail = async (videoId, tag) => {
```
Comment (jontsai) [inline components/tags.js:29]: This function is defined in two files, so it should probably just be defined in a single place (3rd file) instead, and imported in the two file.

This is the very important SWE principle called DRY:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: components/tags.js [hunk 2]
```diff
@@ -1,33 +1,88 @@
...
+        try {
+          const apiUrl = `https://www.youtube.com/oembed?url=https://www.youtube.com/watch?v=${videoId}&format=json`;
```
Comment (jontsai) [inline components/tags.js:31]: I'm not sure if it makes sense to do a live call for each thumbnail, as it adds to the processing time to load the page.

Ideally, we update the Python script used to generate the JSON file, to fetch the thumbnail images at that time.

**Review by jontsai — CHANGES_REQUESTED** [review]

File: components/tags.js (hunk 2 above)
Comment (Pinimes) [reply components/tags.js]: Okay, would it be acceptable to manually insert the generated thumbnail URLs into THEOLOGY101_DATA JSON?

File: components/tags.js (hunk 1 above)
Comment (Pinimes) [reply components/tags.js]: done

**Review by lleung030 — APPROVED** [review]
LFTM

**Review by goztrk — CHANGES_REQUESTED** [review]

File: components/base_layout.jsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ import '@fortawesome/fontawesome-svg-core/styles.css';
 
 import { Footer } from '@components/footer';
 import { Header } from '@components/header';
+import styles from 'styles/common.module.scss';
```
[the remark this answers is not in the document]
Comment (Pinimes) [reply components/base_layout.jsx]: done

File: components/base_layout.jsx [hunk 4]
```diff
@@ -18,16 +19,66 @@ import {
...
+                    ...prevUrls,
+                    [key]: thumbnailForSidebar[videoId].thumbnail,
+                }));
+            }
+        });
+    };
+    fetchThumbnails();
+}
+
 function SidebarList({ title, entries, url }) {
```
[the remark this answers is not in the document]
Comment (Pinimes) [reply components/base_layout.jsx]: done

File: styles/sidebar.module.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
+.sidebar {
+    ul {
+        height: 500px;
```
[the remark this answers is not in the document]
Comment (Pinimes) [reply styles/sidebar.module.scss]: reverted.

**Review by goztrk — CHANGES_REQUESTED** [review]
Code wise it is fine but as UI perspective it is not good enough.

- The video count is not consistent.
- The sidebar is too long. It would be ok If we had that long content but we don't.

**Review by goztrk — COMMENTED** [review]
LFTM for now.

But the design is still not enough. The sidebar is still too long. Could you please file a Trello card and work on it after this PR got merged?

- First come up with design ideas.
  - It does not have to be the same design. Maybe a nice way to show all Topics in the homepage?
  - Check ThemeForest for inspiration.
  - Consider the option of not using Sidebar at all?
  - Or maybe when user opens a video page we can only show videos from the same topic on the sidebar. Just like YouTube Playlists.

**Review by goztrk — APPROVED** [review]

**Review by YosefAshenafi — APPROVED** [review]
LGTM

**Review by jontsai — APPROVED** [review]
LGTM

**Pull request merged**